### ISSUE TITLE
Multi-GPU Training Hangs When Skipping Parameters with `p.grad is None`

### ISSUE THREAD

**Issue by NuanBaobao** [KellerJordan/Muon#37]
Thanks for your work! I am encountering an issue when using the latest code. During the forward pass, I have some parameters that should not participate in gradient computation. To address this, I modified the code as follows:
```python
if p.grad is None:
    continue
```
This works correctly in **single-GPU** mode, but the training **hangs in multi-GPU** (DDP) settings.

## Code Changes
The modified optimizer logic (simplified):
```python
for group in self.param_groups:
    if group["use_muon"]:
        params = group["params"]
        params_pad = params + [torch.empty_like(params[-1])] * (dist.get_world_size() - len(params) % dist.get_world_size())
        for base_i in range(len(params))[::dist.get_world_size()]:
            if base_i + dist.get_rank() < len(params):
                p = params[base_i + dist.get_rank()]
                if p.grad is None:  # Skip non-participating parameters
                    continue  # <--- Causes hang in multi-GPU
                state = self.state[p]
                if len(state) == 0:
                    state["momentum_buffer"] = torch.zeros_like(p)
                update = muon_update(...)
                p.mul_(...)
                p.add_(...)
            dist.all_gather(...)  # Synchronization point
    else:
        for p in group["params"]:
            if p.grad is None:
                continue  # <--- Same issue for non-muon params
            state = self.state[p]
            ...
```
Any help or suggestions would be greatly appreciated!

**Comment by NuanBaobao** [comment]
When using the following code:

```python
if p.grad is None:  # Skip non-participating parameters
    # continue  # <--- Causes hang in multi-GPU
    p.grad = torch.zeros_like(p)  # Force synchronization
```
This issue can be resolved to avoid this hang in multi-GPU environments.

**Comment by KellerJordan** [comment]
should be fixed
